arrow-core-data/src/test/kotlin/arrow/typeclasses/TraverseTest.kt

```diff
@@ -0,0 +1,17 @@
+package arrow.typeclasses
+
+import arrow.core.None
+import arrow.core.Option
+import arrow.core.Some
+import arrow.core.extensions.option.applicative.applicative
+import arrow.core.extensions.sequence.traverse.sequence
+import io.kotlintest.shouldBe
+import io.kotlintest.specs.StringSpec
+
+class TraverseTest : StringSpec({
+  "traverse is stacksafe over very long collections and short circuits properly" {
+    // This has to traverse 50k elements till it reaches None and terminates
+    generateSequence(0) { it + 1 }.map { if (it < 50_000) Some(it) else None }
```

Comment:
Should this be a toggle-able `TraverseLaw` like we do for `ConcurrentLaws` and `MonadDeferLaws` with `testStacksafety` ?

Reply:
Don't think that is wise because this tests the applicative instance more than the traverse instance. This could be a toggleable law in apply which makes sure that `apEval` is stacksafe.

Reply:
But a toggle-able law in apply means a lot of laws need to add a default parameter of sorts to their constructors and apply has a lot of laws above it

Reply:
Doesn't matter too much too me anyway, but we could also make it non-toggleable because it uses `Eval.map` it should be stacksafe anyway, without a manual implementation it just won't short-circuit

Reply:
Alright, makes sense!